src/pages/new-lesson/NewLesson.tsx

```diff
@@ -1,5 +1,142 @@
+import React from 'react'
+import { useTranslation } from 'react-i18next'
+import useForm from '~/hooks/use-form'
+import useAxios from '~/hooks/use-axios'
+import { Link, useNavigate } from 'react-router-dom'
+
+import { Editor } from '@tinymce/tinymce-react'
+import Box from '@mui/material/Box'
+import Divider from '@mui/material/Divider'
+import PageWrapper from '~/components/page-wrapper/PageWrapper'
+import AppTextField from '~/components/app-text-field/AppTextField'
+import AppButton from '~/components/app-button/AppButton'
+
+import { ResourceService } from '~/services/resource-service'
+import { useSnackBarContext } from '~/context/snackbar-context'
+import { AxiosResponse } from 'axios'
+
+import {
+  validations,
+  initialValues,
+  initialFileValue,
+  myResourcesPath
+} from '~/pages/new-lesson/NewLesson.constants'
+import { snackbarVariants } from '~/constants'
+import {
+  ButtonTypeEnum,
+  ButtonVariantEnum,
+  ComponentEnum,
+  ErrorResponse,
+  SizeEnum,
+  NewLessonData,
+  TextFieldVariantEnum
+} from '~/types'
+import { styles } from '~/pages/new-lesson/NewLesson.styles'
+
 const NewLesson = () => {
-  return <h1>New Lesson page</h1>
+  const { t } = useTranslation()
+  const { setAlert } = useSnackBarContext()
+  const navigate = useNavigate()
+
+  const handleResponseError = (error: ErrorResponse) => {
+    setAlert({
+      severity: snackbarVariants.error,
+      message: error ? `errors.${error.code}` : ''
+    })
+  }
+
+  const handleResponse = () => {
+    setAlert({
+      severity: snackbarVariants.success,
+      message: 'newLesson.successMessage'
+    })
+    navigate('/my-resources')
+  }
+
+  const addLesson = (): Promise<AxiosResponse> => {
```

Comment:
use useCallback. Also, I think this Promise<AxiosResponse> is redundant here because you set type in the service

Reply:
i think it is unnecessary to use useCallback here. What are the benefits of this?

Reply:
Okay, but do you need this Promise<AxiosResponse>?